common/logger/src/lib.rs

```diff
@@ -3,131 +3,361 @@
+#[macro_export]
+macro_rules! crit {
+    ($($arg:tt)+) => (
+        $crate::log::error!(target: $crate::DEFAULT_TARGET, $($arg)+);
+    )
+}
+
+/// Define debug macro that specify libra as the target
+// TODO Remove historical crit from code base since it isn't supported in Rust Log.
+#[macro_export]
+macro_rules! debug {
+    ($($arg:tt)+) => (
+        $crate::log::debug!(target: $crate::DEFAULT_TARGET, $($arg)+);
+    )
+}
+
+/// Define  macro that specify libra as the target
+#[macro_export]
+macro_rules! error {
+    ($($arg:tt)+) => (
+        $crate::log::error!(target: $crate::DEFAULT_TARGET, $($arg)+);
+    )
+}
+
+/// Define info macro that specify libra as the target
+#[macro_export]
+macro_rules! info {
+    ($($arg:tt)+) => (
+        $crate::log::info!(target: $crate::DEFAULT_TARGET, $($arg)+);
+    )
+}
+
+/// Define trace macro that specify libra as the target
+#[macro_export]
+macro_rules! trace {
+    ($($arg:tt)+) => (
+        $crate::log::trace!(target: $crate::DEFAULT_TARGET, $($arg)+);
+    )
+}
+
+/// Define warn macro that specify libra as the target
+#[macro_export]
+macro_rules! warn {
+    ($($arg:tt)+) => (
+        $crate::log::warn!(target: $crate::DEFAULT_TARGET, $($arg)+);
+    )
+}
+
+/// Logging framework for Libra that encapsulates a minimal dependency logger with support for
+/// environmental variable (RUST_LOG) and asynchronous logging.
+/// Note: only a single logger can be instantiated at a time. Repeated instantiates of the loggers
+/// will only adjust the global logging level but will not change the initial filter.
+pub struct Logger {
+    /// Channel size for sending logs to async handler.
+    channel_size: usize,
+    /// Only instantiate a logger if the environment is properly set
+    environment_only: bool,
+    /// Use a dedicated thread for logging.
+    is_async: bool,
+    /// The default logging level.
+    level: Level,
+    /// Override RUST_LOG even if set
+    override_rust_log: bool,
+}
+
+impl Logger {
+    pub fn new() -> Self {
+        Self {
+            channel_size: CHANNEL_SIZE,
+            environment_only: false,
+            is_async: false,
+            level: Level::Info,
+            override_rust_log: false,
+        }
+    }
+
+    pub fn channel_size(&mut self, channel_size: usize) -> &mut Self {
+        self.channel_size = channel_size;
+        self
+    }
+
+    pub fn environment_only(&mut self, environment_only: bool) -> &mut Self {
+        self.environment_only = environment_only;
+        self
+    }
+
+    pub fn is_async(&mut self, is_async: bool) -> &mut Self {
+        self.is_async = is_async;
+        self
+    }
+
+    pub fn level(&mut self, level: Level) -> &mut Self {
+        self.level = level;
+        self
+    }
+
+    pub fn override_rust_log(&mut self, override_rust_log: bool) -> &mut Self {
+        self.override_rust_log = override_rust_log;
+        self
+    }
+
+    pub fn init(&mut self) {
+        self.internal_init(StdoutWriter {});
```

Comment:
I believe we were writing to stderr before - could you fix this so it writes to stderr?
Panics and other things are going there, it would be nice if we have won't have unpredictable interleaving there

Reply:
verified working in cluster test!
good call, back to sdterr!